### PR TITLE
Ood: Yet Another Yaml Processor

### PR DESCRIPTION
**Pull request by cuihtlauac**
- Ood: Extract Yaml Decoding functions (#868)
- Result monad style
- Streamlining
- Spread
- Use Result in monad style
- Remove redefinition of Result.fold

### CODE REVIEW

**Review by sabine — APPROVED** [review]
Less noise in the individual files, so that's good!

File: tool/ood-gen/lib/utils.ml [hunk 1]
```diff
@@ -28,30 +24,47 @@ let decode_or_raise f x =
...
+  let key_default = Filename.(file |> basename |> remove_extension) in
+  let key = Option.value ~default:key_default key in
+  (let* yaml = yaml_file file in
+   let* opt = Yaml.Util.find key yaml in
+   let* found = Option.to_result ~none:(`Msg (key ^ ", key not found")) opt in
+   let* list =
+     (function `A u -> Ok u | _ -> Error (`Msg "expecting a sequence")) found
+   in
+   List.fold_left (fun u x -> Ok List.cons <@> of_yaml x <@> u) (Ok []) list)
+  |> Result.map_error (function `Msg err -> `Msg (file ^ ": " ^ err))
```
Comment (sabine) [inline tool/ood-gen/lib/utils.ml:65]: Maybe here?

Comment (cuihtlauac) [reply tool/ood-gen/lib/utils.ml:65]: It's the other one ;-)

**Comment by cuihtlauac** [comment]
Added a commit fixing the order reversal issue

**Pull request merged**